_troubleshooting/cloud/hypershift-bad-perf.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Hypershift migration runs slowly
```

Comment:
Unfortunately there are multiple issues that manifest from this failure mode, not just "slowness".

What we want hypershift to do is to copy multiple parts of the source table in parallel (essentially chunks, but plain postgres doesn't have chunks). When compression is enabled on the destination table, we can compress each chunk immediately after copying, this is how we avoid peak storage usage.

Without an index, this parallel copy is not performed. This means that not only is the copy slower (because we couldn't copy in parallel), but the target DB instance will require the same amount of storage for the table as the source database did, at least until compression is applied.

I know this is a lot, and goes kind of deep into how Timescale and Hypershift works, but IMHO it's important that we explain why these effects happen.

Reply:
OK, I'm happy to include more detail in the entry, but we can probably leave the title as is?

Reply:
Ya, sure, that works for me!